### PR TITLE
WIP bumps comscore to v5

### PR DESCRIPTION
**Pull request by ladanazita**
Hit some blocks here, mainly unsure of how to set `usagePropertiesAutoUpdateMode` to `ForegroundOnly`, `ForegroundAndBackground`, or `Disabled` in builder method

why getApplicationContext isn't found
`com.comscore.Analytics.start(this.getApplicationContext());`

comScore is also named `Analytics` so for example, instead of `Analytics.getConfiguration().addClient(myPublisherConfig);` I specify `com.comscore.Analytics.getConfiguration().addClient(myPublisherConfig);`

Also need to confirm w/ comScore on screen and track as it is not documented. I could see the more similar syntax for hidden -> was notifyHiddenEvent and view -> notifyViewEvent.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -27,20 +27,36 @@ android {
...
 }
 
 checkstyle {
   configFile rootProject.file('gradle/checkstyle.xml')
 }
 
+allprojects {
+  repositories {
+    maven {
+      url "https://comscore.bintray.com/Analytics"
```
Comment (f2prateek) [inline build.gradle]: We should add this to the Readme. Customers will need to add this block to their Gradle configuration.

File: build.gradle [hunk 2]
```diff
@@ -27,20 +27,36 @@ android {
...
+
 dependencies {
   repositories {
     mavenCentral()
   }
 
   provided 'com.segment.analytics.android:analytics:4.0.0'
 
-  compile files('libs/comscore.jar')
+  compile "com.comscore:android-analytics:5.+"
```
Comment (f2prateek) [inline build.gradle]: Single quotes for consistency.

File: build.gradle [hunk 3]
```diff
@@ -61,4 +77,4 @@ dependencies {
   testCompile 'org.powermock:powermock-classloading-xstream:1.6.2'
 }
 
-apply from: rootProject.file('gradle/gradle-mvn-push.gradle')
+apply from: rootProject.file('gradle/gradle-mvn-push.gradle')
```
Comment (f2prateek) [inline build.gradle]: Should end with a newline.

File: src/main/java/com/segment/analytics/android/integrations/comscore/ComScoreIntegration.java [hunk 4]
```diff
@@ -1,18 +1,21 @@
...
+import com.comscore.Analytics;
+import com.comscore.PublisherConfiguration;
+import com.comscore.UsagePropertiesAutoUpdateMode;
 import com.segment.analytics.ValueMap;
 import com.segment.analytics.integrations.IdentifyPayload;
 import com.segment.analytics.integrations.Integration;
 import com.segment.analytics.integrations.Logger;
 import com.segment.analytics.integrations.ScreenPayload;
 import com.segment.analytics.integrations.TrackPayload;
+
```
Comment (f2prateek) [inline src/main/java/com/segment/analytics/android/integrations/comscore/ComScoreIntegration.java]: unnecessary space.

File: src/main/java/com/segment/analytics/android/integrations/comscore/ComScoreIntegration.java [hunk 5]
```diff
@@ -41,36 +45,46 @@
...
+        );
+      } else if (foregroundOnly) {
+        builder.usagePropertiesAutoUpdateMode(UsagePropertiesAutoUpdateMode.FOREGROUND_ONLY);
       } else {
-        comScore.enableAutoUpdate(autoUpdateInterval, false);
+        builder.usagePropertiesAutoUpdateMode(UsagePropertiesAutoUpdateMode.DISABLED);
       }
-    }
+
+      PublisherConfiguration myPublisherConfig = builder.build();
```
Comment (f2prateek) [inline src/main/java/com/segment/analytics/android/integrations/comscore/ComScoreIntegration.java]: Needs to be run through the formatter.

File: src/main/java/com/segment/analytics/android/integrations/comscore/ComScoreIntegration.java [hunk 6]
```diff
@@ -41,36 +45,46 @@
...
+      Analytics.start(analytics.getApplication());
   }
 
   @Override public void track(TrackPayload track) {
     String name = track.event();
     HashMap<String, String> properties = (HashMap<String, String>) track.properties().toStringMap();
     properties.put("name", name);
-    comScore.hidden(properties);
-    logger.verbose("comScore.hidden(%s)", properties);
+      Analytics.notifyHiddenEvent(properties);
```
Comment (f2prateek) [inline src/main/java/com/segment/analytics/android/integrations/comscore/ComScoreIntegration.java]: Formatter error.

File: src/main/java/com/segment/analytics/android/integrations/comscore/ComScoreIntegration.java [hunk 7]
```diff
@@ -81,8 +95,8 @@
     properties.put("name", name);
     properties.put("category", category);
 
-    comScore.view(properties);
-    logger.verbose("comScore.hidden(%s)", properties);
+      Analytics.notifyViewEvent(properties);
```
Comment (f2prateek) [inline src/main/java/com/segment/analytics/android/integrations/comscore/ComScoreIntegration.java]: Formatter error.

File: src/test/java/com/segment/analytics/android/integrations/comscore/ComScoreTest.java [hunk 8]
```diff
@@ -1,24 +1,26 @@
 package com.segment.analytics.android.integrations.comscore;
 
-
 import android.app.Application;
-import com.comscore.analytics.comScore;
-import com.segment.analytics.Analytics;
+
```
Comment (f2prateek) [inline src/test/java/com/segment/analytics/android/integrations/comscore/ComScoreTest.java]: Unneessary space.

File: src/test/java/com/segment/analytics/android/integrations/comscore/ComScoreTest.java [hunk 9]
```diff
@@ -145,18 +159,18 @@
...
 
     LinkedHashMap<String, String> expected = new LinkedHashMap<>();
     expected.put("name", "SmartWatches");
     expected.put("category", "Purchase Screen");
 
     verifyStatic();
-    comScore.view(expected);
+    Analytics.notifyViewEvent(expected);
   }
 }
```
Comment (f2prateek) [inline src/test/java/com/segment/analytics/android/integrations/comscore/ComScoreTest.java]: Should end with a newline.

**Comment by f2prateek** [comment]
LGTM, just needs some style changes. Running it through the formatter will fix almos all if not most of them.

File: build.gradle (hunk 1 above)
Comment (ladanazita) [inline build.gradle]: done!

File: build.gradle [hunk 10]
```diff
@@ -27,20 +29,36 @@ android {
...
+
 dependencies {
   repositories {
     mavenCentral()
   }
 
   provided 'com.segment.analytics.android:analytics:4.0.0'
 
-  compile files('libs/comscore.jar')
+  compile 'com.comscore:android-analytics:5.+'
```
Comment (f2prateek) [inline build.gradle]: Actually this should be a specific version.

**Comment by ladanazita** [comment]
Tested with Charles and see the following coming through:

```
c1=19
c2=24186693
ns_ap_an=ladanAndroidTestApp
ns_ap_pn=android
ns_ap_pv=6.0
c12=f0aeafa8e074f0f97cb86667cf88b657-cs72
ns_ak=PM5LpWZxV3fV8%2FNEO%2FlE8ZfuyvjmXSGF%2FyVKuQPQvgXlgybk9Vw12Xv7ckriBF84LK3v2fF39FzCX8oPIQXhflPJu2hy1%2F4PFuS49vDul7eI1ehPyT2w6VV0wvx8cIaO6PtfImrt4LUTDcLtIkcudfJLUd3XwTKt5ZpTEDYTNjM%3D
name=Button%20A%20Clicked
ns_ap_ec=5
ns_ap_ev=hidden
ns_ap_device=vbox86p
ns_ap_id=1490648849690
ns_ap_bi=com.segment.analytics.sample
ns_ap_pfm=android
ns_ap_pfv=6.0
ns_ap_ver=4.2.6
ns_ap_sv=5.1.0.170220
ns_ap_smv=2.11
ns_type=hidden
ns_radio=wifi
ns_nc=1
ns_ap_res=1080x1776
ns_ap_sd=1080x1920
ns_ap_cs=2
ns_ap_usage=968593
ns_ap_fg=0
ns_ap_ft=147651
ns_ap_dft=3613
ns_ap_bt=0
ns_ap_dbt=0
ns_ap_dit=0
ns_ap_as=0
ns_ap_das=3613
ns_ap_it=0
ns_ap_ut=60000
ns_ap_lang=en
ns_ap_ar=i686
ns_ts=1490648997341
ns_ap_i3=8acad5140a2f3319e688a75e50c0ed6a 
```

For now, since the comScore UI doesn't show any granular details, I will assume the above is sufficient to say that everything is working as expected.

**Pull request merged**